wren-ui/src/apollo/server/backgrounds/dashboardCacheBackgroundTracker.ts

```diff
@@ -0,0 +1,194 @@
+  DashboardCacheRefreshStatus,
+} from '@server/repositories';
+import {
+  IProjectService,
+  IDeployService,
+  IQueryService,
+} from '@server/services';
+import { CronExpressionParser } from 'cron-parser';
+import { v4 as uuidv4 } from 'uuid';
+
+const logger = getLogger('DashboardCacheBackgroundTracker');
+logger.level = 'debug';
+
+export class DashboardCacheBackgroundTracker {
+  private intervalTime: number;
+  private dashboardRepository: IDashboardRepository;
+  private dashboardItemRepository: IDashboardItemRepository;
+  private dashboardCacheRefreshRepository: IDashboardCacheRefreshRepository;
+  private projectService: IProjectService;
+  private deployService: IDeployService;
+  private queryService: IQueryService;
+  private runningJobs = new Set<number>();
+
+  constructor({
+    dashboardRepository,
+    dashboardItemRepository,
+    dashboardCacheRefreshRepository,
+    projectService,
+    deployService,
+    queryService,
+  }: {
+    dashboardRepository: IDashboardRepository;
+    dashboardItemRepository: IDashboardItemRepository;
+    dashboardCacheRefreshRepository: IDashboardCacheRefreshRepository;
+    projectService: IProjectService;
+    deployService: IDeployService;
+    queryService: IQueryService;
+  }) {
+    this.dashboardRepository = dashboardRepository;
+    this.dashboardItemRepository = dashboardItemRepository;
+    this.dashboardCacheRefreshRepository = dashboardCacheRefreshRepository;
+    this.projectService = projectService;
+    this.deployService = deployService;
+    this.queryService = queryService;
+    this.intervalTime = 60000; // 1 minute
+    this.start();
+  }
+
+  private start(): void {
+    logger.info('Dashboard cache background tracker started');
+    setInterval(() => {
+      this.checkAndRefreshCaches();
+    }, this.intervalTime);
+  }
+
+  private async checkAndRefreshCaches(): Promise<void> {
+    try {
+      // Get all dashboards with cache enabled
+      const dashboards = await this.dashboardRepository.findAllBy({
+        cacheEnabled: true,
+      });
+
+      for (const dashboard of dashboards) {
+        if (!dashboard.scheduleCron || !dashboard.nextScheduledAt) {
+          continue;
+        }
+
+        const now = new Date();
+        const nextScheduledAt = new Date(dashboard.nextScheduledAt);
+
+        // Check if it's time to refresh
+        if (now >= nextScheduledAt) {
+          logger.info(`Start Refreshing cache for dashboard ${dashboard.id}`);
+          await this.refreshDashboardCache(dashboard);
+          logger.info(
+            `Finished Refreshing cache for dashboard ${dashboard.id}`,
+          );
+        }
+      }
+    } catch (error) {
+      logger.error(`Error checking dashboard caches: ${error.message}`);
+    }
+  }
+
+  private async refreshDashboardCache(dashboard: any): Promise<void> {
+    if (this.runningJobs.has(dashboard.id)) {
+      logger.debug(`Dashboard ${dashboard.id} refresh already in progress`);
```

Comment:
I'm thinking it's possible our logs will be bombed with this message, since the check is per minute ?

Reply:
Setting production env logging level to `info` should be good enough for now.